spec/api.md

```diff
@@ -0,0 +1,129 @@
+# CAS API Documentation
+
+This document describes the HTTP API endpoints used by the CAS (Content Addressable Storage) client to interact with the remote CAS server.
+
+## Authentication
+
+- TODO
+
+## Endpoints
+
+### 1. Get File Reconstruction
+
+**Description**: Retrieves reconstruction information for a specific file, optionally with byte range support.
+
+- **Path**: `/v1/reconstruction/{file_id}`
+- **Method**: `GET`
+- **Parameters**:
+  - `file_id`: MerkleHash in hex format
+- **Headers**:
+  - `Range`: Optional. Format: `bytes={start}-{end}` (inclusive end)
+- **Body**: None
+- **Response**: JSON (`QueryReconstructionResponse`)
+
+  ```json
+  {
+    "offset_into_first_range": 0,
+    "terms": [...],
+    "fetch_info": {...}
+  }
+  ```
+
+- **Error Responses**:
+  - 400 bad request
+  - 404 File not found
+  - 416 Range Not Satisfiable: When requested byte range start exceeds the end of a file
+
+### 2. Batch Get Reconstruction
+
+**Description**: Retrieves reconstruction information for multiple files in a single request.
+
+- **Path**: `/v1/reconstructions`
+- **Method**: `GET`
+- **Query Parameters**:
+  - `file_id`: MerkleHash(es) in hex format (can be repeated)
+  - Example: `/v1/reconstructions?file_id=abc123&file_id=def456`
+- **Headers**: None (beyond authentication)
+- **Body**: None
+- **Response**: JSON (`BatchQueryReconstructionResponse`)
+
+- **Error Responses**:
+  - 400 bad request
+  - 404 File not found if any file is not found
+
+### 3. Query Chunk Deduplication (Global Deduplication)
+
+**Description**: Checks if a chunk exists in the CAS for deduplication purposes.
+
+- **Path**: `/v1/chunk/default-merkledb/{hash}`
```

Comment:
What purpose is the default here? I would expect this to be `GET /v1/chunks/{hash}` unless `default` is variable.